superset-frontend/src/dashboard/components/FiltersBadge/selectors.ts

```diff
@@ -0,0 +1,151 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { isNil, get } from 'lodash';
+import { getChartIdsInFilterScope } from '../../util/activeDashboardFilters';
+import { TIME_FILTER_MAP } from '../../../visualizations/FilterBox/FilterBox';
+
+export enum IndicatorStatus {
+  Unset = 'UNSET',
+  Applied = 'APPLIED',
+  Incompatible = 'INCOMPATIBLE',
+}
+
+const TIME_GRANULARITY_FIELDS = new Set([
```

Comment:
For Druid datasources, it has different time column and granularity keys. why not take care of them?